internal/auth/auth.go

```diff
@@ -59,27 +61,45 @@ func (sarag *serviceAccountRegistryAuthGetter) GetKeyChain(ctx context.Context)
 }
 
 type RegistryAuthGetterFactory interface {
-	NewRegistryAuthGetter(client client.Client, namespacedName types.NamespacedName) RegistryAuthGetter
-	NewServiceAccountRegistryAuthGetter(coreClientSet k8s.Interface, namespace, serviceAccountName string) RegistryAuthGetter
+	NewRegistryAuthGetterFrom(mod *kmmv1beta1.Module) RegistryAuthGetter
 }
 
-type registryAuthGetterFactory struct{}
+type registryAuthGetterFactory struct {
+	client        client.Client
+	coreClientSet k8s.Interface
+}
 
-func NewRegistryAuthGetterFactory() RegistryAuthGetterFactory {
-	return registryAuthGetterFactory{}
+func NewRegistryAuthGetterFactory(client client.Client, coreClientSet k8s.Interface) RegistryAuthGetterFactory {
+	return &registryAuthGetterFactory{
+		client:        client,
+		coreClientSet: coreClientSet,
+	}
 }
 
-func (registryAuthGetterFactory) NewRegistryAuthGetter(client client.Client, namespacedName types.NamespacedName) RegistryAuthGetter {
+func (af *registryAuthGetterFactory) newRegistryAuthGetter(namespacedName types.NamespacedName) RegistryAuthGetter {
 	return &registrySecretAuthGetter{
-		client:         client,
+		client:         af.client,
 		namespacedName: namespacedName,
 	}
 }
 
-func (registryAuthGetterFactory) NewServiceAccountRegistryAuthGetter(coreClientSet k8s.Interface, namespace, serviceAccountName string) RegistryAuthGetter {
+func (af *registryAuthGetterFactory) newServiceAccountRegistryAuthGetter(namespace, serviceAccountName string) RegistryAuthGetter {
 	return &serviceAccountRegistryAuthGetter{
-		coreClientSet:      coreClientSet,
+		coreClientSet:      af.coreClientSet,
 		namespace:          namespace,
 		serviceAccountName: serviceAccountName,
 	}
 }
+
+func (af *registryAuthGetterFactory) NewRegistryAuthGetterFrom(mod *kmmv1beta1.Module) RegistryAuthGetter {
+	if mod.Spec.ImageRepoSecret != nil {
+		namespacedName := types.NamespacedName{
+			Name:      mod.Spec.ImageRepoSecret.Name,
+			Namespace: mod.Namespace,
+		}
+		return af.newRegistryAuthGetter(namespacedName)
+	}
+	return af.newServiceAccountRegistryAuthGetter(
+		mod.Namespace,
+		constants.OCPBuilderServiceAccountName)
```

Comment:
```suggestion
	return af.newServiceAccountRegistryAuthGetter(
		mod.Namespace,
		constants.OCPBuilderServiceAccountName
		)
```

Reply:
i run both linter and go fmt, and it was happy with the code. Where is the convention for moving the brace to the next line?

Reply:
I though that this is better, but `go fmt` is indeed fixing it.
Let's keep it that way.